sampler.go

```diff
@@ -0,0 +1,71 @@
+// Licensed to SkyAPM org under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. SkyAPM org licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package go2sky
+
+import (
+	"math/rand"
+	"time"
+)
+
+type Sampler interface {
+	IsSampled(traceID string, operation string) (sampled bool)
+}
+
+type ConstSampler struct {
+	decision bool
+}
+
+// NewConstSampler creates a ConstSampler.
+func NewConstSampler(sample bool) *ConstSampler {
+	s := &ConstSampler{
+		decision: sample,
+	}
+	return s
+}
+
+// IsSampled implements IsSampled() of Sampler.
+func (s *ConstSampler) IsSampled(traceID string, operation string) bool {
+	return s.decision
+}
+
+type RandomSampler struct {
+	samplingRate float64
+	rand         *rand.Rand
+	Threshold    int
```

Comment:
Why `T` in upper case?

Reply:
This is just for test, random sampler is not easy to test the correctness

Reply:
changed Threshold to threshold, sorry I am newer to golang